mp/src/game/server/env_player_surface_trigger.cpp

```diff
@@ -67,18 +67,22 @@ void CEnvPlayerSurfaceTrigger::OnRestore( void )
 //-----------------------------------------------------------------------------
 // Purpose: 
 //-----------------------------------------------------------------------------
-void CEnvPlayerSurfaceTrigger::SetPlayerSurface( CBasePlayer *pPlayer, char gameMaterial )
+void CEnvPlayerSurfaceTrigger::SetPlayerSurface( CBasePlayer *pPlayer, const trace_t &tr )
 {
-	// Ignore players in the air (stops bunny hoppers escaping triggers)
-	if ( gameMaterial == 0 )
-		return;
-
-	// Loop through the surface triggers and tell them all about the change
-	int iCount = g_PlayerSurfaceTriggers.Count();
-	for ( int i = 0; i < iCount; i++ )
-	{
-		g_PlayerSurfaceTriggers[i]->PlayerSurfaceChanged( pPlayer, gameMaterial );
-	}
+    int iCount = g_PlayerSurfaceTriggers.Count();
+
+    if (iCount > 0)
```

Comment:
Is this check here necessary? Seems the `for` below checks it. If anything, could be better as an early return.

Reply:
I didn't want it to execute the next 3 lines pointlessly